src/amber/validators/params.cr

```diff
@@ -31,7 +31,12 @@ module Amber::Validators
     end
 
     private def call_predicate(params : Amber::Router::Params)
-      @value = params[@field]
+      begin
```

Comment:
Instead of a `begin`/`rescue` here we can use a `case` statement, pass in the request object or some of the information about the request (specifically the `content-type` value if it's present) so we can determine if we need to parse for JSON or not.

Reply:
That is handled earlier up the stack. The parameters passed into the validator are all the body params and the route and any other form of param concatenated together. I like the idea of using a `case statment`.

Reply:
Actually, we can't use a case statement because JSON.parse raises and there is no other way to handle it.